Build with Cabana Master fails

```
2020-08-03T02:19:08.1445642Z [ 66%] Building CXX object examples/CMakeFiles/FreeFall.dir/free_fall.cpp.o
2020-08-03T02:19:08.1455217Z cd /tmp/runner/spack-stage/spack-stage-exampm-master-n5v37glhooiigpsoumjmjzgpb55kvlw2/spack-build/examples && /home/runner/spack/lib/spack/env/gcc/g++  -I/tmp/runner/spack-stage/spack-stage-exampm-master-n5v37glhooiigpsoumjmjzgpb55kvlw2/spack-build/examples -I/tmp/runner/spack-stage/spack-stage-exampm-master-n5v37glhooiigpsoumjmjzgpb55kvlw2/spack-src/examples -I/tmp/runner/spack-stage/spack-stage-exampm-master-n5v37glhooiigpsoumjmjzgpb55kvlw2/spack-src/src -I/tmp/runner/spack-stage/spack-stage-exampm-master-n5v37glhooiigpsoumjmjzgpb55kvlw2/spack-build/src -isystem /home/runner/spack/opt/spack/linux-ubuntu18.04-skylake/gcc-9.3.0/kokkos-3.1.01-4qzdxlx5fux2ydikb2ez5e42qtp66uhy/include -isystem /home/runner/spack/opt/spack/linux-ubuntu18.04-skylake/gcc-9.3.0/openmpi-3.1.6-5gsagjujszo4s347uw7vkyqn2teqzprv/include -isystem /home/runner/spack/opt/spack/linux-ubuntu18.04-skylake/gcc-9.3.0/cabana-master-xps3be3gaxfbk4izrjvn2hwxz233wfee/include -isystem /home/runner/spack/opt/spack/linux-ubuntu18.04-skylake/gcc-9.3.0/silo-4.10.2-va4cvfzv3ydoyjxjshzdlby4hihpmikt/include -O2 -g -DNDEBUG -march=core-avx2 -mtune=core-avx2 -mrtm -pthread -o CMakeFiles/DamBreak.dir/dam_break.cpp.o -c /tmp/runner/spack-stage/spack-stage-exampm-master-n5v37glhooiigpsoumjmjzgpb55kvlw2/spack-src/examples/dam_break.cpp
2020-08-03T02:19:08.1462512Z cd /tmp/runner/spack-stage/spack-stage-exampm-master-n5v37glhooiigpsoumjmjzgpb55kvlw2/spack-build/examples && /home/runner/spack/lib/spack/env/gcc/g++  -I/tmp/runner/spack-stage/spack-stage-exampm-master-n5v37glhooiigpsoumjmjzgpb55kvlw2/spack-build/examples -I/tmp/runner/spack-stage/spack-stage-exampm-master-n5v37glhooiigpsoumjmjzgpb55kvlw2/spack-src/examples -I/tmp/runner/spack-stage/spack-stage-exampm-master-n5v37glhooiigpsoumjmjzgpb55kvlw2/spack-src/src -I/tmp/runner/spack-stage/spack-stage-exampm-master-n5v37glhooiigpsoumjmjzgpb55kvlw2/spack-build/src -isystem /home/runner/spack/opt/spack/linux-ubuntu18.04-skylake/gcc-9.3.0/kokkos-3.1.01-4qzdxlx5fux2ydikb2ez5e42qtp66uhy/include -isystem /home/runner/spack/opt/spack/linux-ubuntu18.04-skylake/gcc-9.3.0/openmpi-3.1.6-5gsagjujszo4s347uw7vkyqn2teqzprv/include -isystem /home/runner/spack/opt/spack/linux-ubuntu18.04-skylake/gcc-9.3.0/cabana-master-xps3be3gaxfbk4izrjvn2hwxz233wfee/include -isystem /home/runner/spack/opt/spack/linux-ubuntu18.04-skylake/gcc-9.3.0/silo-4.10.2-va4cvfzv3ydoyjxjshzdlby4hihpmikt/include -O2 -g -DNDEBUG -march=core-avx2 -mtune=core-avx2 -mrtm -pthread -o CMakeFiles/FreeFall.dir/free_fall.cpp.o -c /tmp/runner/spack-stage/spack-stage-exampm-master-n5v37glhooiigpsoumjmjzgpb55kvlw2/spack-src/examples/free_fall.cpp
2020-08-03T02:19:09.5645222Z In file included from /tmp/runner/spack-stage/spack-stage-exampm-master-n5v37glhooiigpsoumjmjzgpb55kvlw2/spack-src/src/ExaMPM_Solver.hpp:15,
2020-08-03T02:19:09.5804759Z                  from /tmp/runner/spack-stage/spack-stage-exampm-master-n5v37glhooiigpsoumjmjzgpb55kvlw2/spack-src/examples/free_fall.cpp:1:
2020-08-03T02:19:09.5805597Z /tmp/runner/spack-stage/spack-stage-exampm-master-n5v37glhooiigpsoumjmjzgpb55kvlw2/spack-src/src/ExaMPM_ProblemManager.hpp:78:49: error: wrong number of template arguments (2, should be 1)
2020-08-03T02:19:09.5806127Z    78 |     using halo = Cajita::Halo<double,MemorySpace>;
2020-08-03T02:19:09.5806375Z       |                                                 ^
```
from https://github.com/proxyapps/proxy-ci/runs/938692143?check_suite_focus=true

--- a/src/ExaMPM_ProblemManager.hpp
+++ b/src/ExaMPM_ProblemManager.hpp
@@ -56,6 +56,9 @@ class ProblemManager
 {
   public:
 
+    using memory_space = MemorySpace;
+    using execution_space = typename memory_space::execution_space;
+
     using particle_members = Cabana::MemberTypes<double[3][3],
                                                  double[3],
                                                  double[3],
@@ -75,7 +78,7 @@ class ProblemManager
                                      Cajita::UniformMesh<double>,
                                      MemorySpace>;
 
-    using halo = Cajita::Halo<double,MemorySpace>;
+    using halo = Cajita::Halo<MemorySpace>;
 
     using mesh_type = Mesh<MemorySpace>;
 
@@ -243,42 +246,54 @@ class ProblemManager
 
     void scatter( Location::Node, Field::Momentum ) const
     {
-        _node_vector_halo->scatter( *_momentum );
+        _node_vector_halo->scatter( execution_space(),
+                                    Cajita::ScatterReduce::Sum(),
+                                    *_momentum );
     }
 
     void scatter( Location::Node, Field::Mass ) const
     {
-        _node_scalar_halo->scatter( *_mass );
+        _node_scalar_halo->scatter( execution_space(),
+                                    Cajita::ScatterReduce::Sum(),
+                                    *_mass );
     }
 
     void scatter( Location::Node, Field::Force ) const
     {
-        _node_vector_halo->scatter( *_force );
+        _node_vector_halo->scatter( execution_space(),
+                                    Cajita::ScatterReduce::Sum(),
+                                    *_force );
     }
 
     void scatter( Location::Node, Field::PositionCorrection ) const
     {
-        _node_vector_halo->scatter( *_position_correction );
+        _node_vector_halo->scatter( execution_space(),
+                                    Cajita::ScatterReduce::Sum(),
+                                    *_position_correction );
     }
 
     void scatter( Location::Cell, Field::Density ) const
     {
-        _cell_scalar_halo->scatter( *_density );
+        _cell_scalar_halo->scatter( execution_space(),
+                                    Cajita::ScatterReduce::Sum(),
+                                    *_density );
     }
 
     void scatter( Location::Cell, Field::Mark ) const
     {
-        _cell_scalar_halo->scatter( *_mark );
+        _cell_scalar_halo->scatter( execution_space(),
+                                    Cajita::ScatterReduce::Sum(),
+                                    *_mark );
     }
 
     void gather( Location::Node, Field::Velocity ) const
     {
-        _node_vector_halo->gather( *_velocity );
+        _node_vector_halo->gather( execution_space(), *_velocity );
     }
 
     void gather( Location::Node, Field::PositionCorrection ) const
     {
-        _node_vector_halo->gather( *_position_correction );
+        _node_vector_halo->gather( execution_space(), *_position_correction );
     }
 
     void communicateParticles( const int minimum_halo_width )

